hangman.py

```diff
@@ -83,63 +77,170 @@ def get_available_letters(letters_guessed):
-    
+
     * Ask the user to supply one guess per round. Remember to make
       sure that the user puts in a letter!
-    
-    * The user should receive feedback immediately after each guess 
+
+    * The user should receive feedback immediately after each guess
       about whether their guess appears in the computer's word.
 
-    * After each guess, you should display to the user the 
+    * After each guess, you should display to the user the
       partially guessed word so far.
-    
+
     Follows the other limitations detailed in the problem write-up.
     '''
-    # FILL IN YOUR CODE HERE AND DELETE "pass"
-    pass
-
-
-
-# When you've completed your hangman function, scroll down to the bottom
-# of the file and uncomment the first two lines to test
-#(hint: you might want to pick your own
-# secret_word while you're doing your own testing)
-
-
-# -----------------------------------
-
+    letters_guessed = []
+    guesses = 0
+    warnings = 0
+    unique_letters = ''
+    for letter in secret_word:
+        if letter not in unique_letters:
+            unique_letters += letter
+
+    print('Welcome to the game Hangman!')
+    print('I\'m thinking of a word that is ' + str(len(secret_word)) + ' letters long.')
+
+    while guesses < 6:
+        letters_left = get_available_letters(letters_guessed)
+        print('------------')
+        print('You have ' + str(6 - guesses) + ' guesses left.')
+        print('Available letters: ' + letters_left)
+
+        user_guess = (input('Please guess a letter: '))
+
+        if str.isalpha(user_guess):
+            if str.isupper(user_guess):
+                user_guess = str.lower(user_guess)
+            if user_guess in letters_guessed:
+                warnings += 1
+                if warnings < 4:
+                    print('Oops! You\'ve already guessed that letter.You now have ' + str(3 - warnings) +
+                          ' warnings left:' + get_guessed_word(secret_word, letters_guessed))
+                elif warnings == 4:
+                    print(
+                        'Oops! You\'ve already guessed that letter. You now have no warnings left so you lose one guess:'
+                        + get_guessed_word(secret_word, letters_guessed))
+                    guesses += 1
+                else:
+                    break
+
+            else:
+                letters_guessed.append(user_guess)
+                if is_word_guessed(user_guess, secret_word):
+                    print('Good guess:', get_guessed_word(secret_word, letters_guessed))
+                    if get_guessed_word(secret_word, letters_guessed) == secret_word:
+                        print('------------\nCongratulations, you won!\n'
+                              'Your total score for this game is:' + (len(set(secret_word)) * (6 - guesses)))
+                        break
+
+                else:
+                    if user_guess in ['a', 'e', 'i', 'o', 'u']:
```

Comment:
Add vowels to set and declare it as VOWELS in the top of file, it must be constant.